### PR TITLE
[Python] Remove Six dependency

### PR DESCRIPTION
**Pull request by XuanWang-Amos**
### Description

Remove `six` dependency since now we only support Python3.

### Changes included
* Add `build/` to `.gitignore`.
* Replace `six.itervalues()` and `six.iterkeys()` with `values()` and `keys()`.
* Replace `six.iteritems()` with `items()`.
* Replace `six.assertCountEqual()` with `assertCountEqual()`.
* Remove unnecessary `import`.

### Testing
* Failed Tests:
  * Basic Tests C++ iOS
    * Also [failing in master](https://fusion.corp.google.com/projectanalysis/summary/KOKORO/prod:grpc%2Fcore%2Fmaster%2Fmacos%2Fgrpc_basictests_cpp_ios).
  * Basic Tests ObjC iOS
    * Also [failing in master](https://fusion.corp.google.com/projectanalysis/summary/KOKORO/prod:grpc%2Fcore%2Fmaster%2Fmacos%2Fgrpc_basictests_objc_ios).

### CODE REVIEW

File: tools/run_tests/python_utils/report_utils.py [hunk 1]
```diff
@@ -24,9 +24,6 @@
 import string
 import xml.etree.cElementTree as ET
 
-import six
```
Comment (gnossen) [inline tools/run_tests/python_utils/report_utils.py:27]: No need to work on the Python scripts under `tools/run_tests`. This is just infrastructure code and, more problematically, there aren't good unit tests for it.

**Comment by gnossen** [comment]
@XuanWang-Amos Just want to make sure I understand the state of this PR. It's still in a draft state, but you've assigned me as a reviewer and an assignee. Should I review now or no?

**Comment by XuanWang-Amos** [comment]
> @XuanWang-Amos Just want to make sure I understand the state of this PR. It's still in a draft state, but you've assigned me as a reviewer and an assignee. Should I review now or no?

Don't need review it now since I'm still working on this (I thought it won't ask for review if it's still in draft).

File: tools/run_tests/python_utils/report_utils.py (hunk 1 above)
Comment (XuanWang-Amos) [reply tools/run_tests/python_utils/report_utils.py]: Removed changes to `tools/*`.

**Review by gnossen — APPROVED** [review]
👍

**Pull request merged**